lib/mocha/mock.rb

```diff
@@ -305,11 +305,11 @@ def method_missing(symbol, *arguments, &block)
         raise NoMethodError, "undefined method `#{symbol}' for #{mocha_inspect} which responds like #{@responder.mocha_inspect}"
       end
       if (matching_expectation_allowing_invocation = all_expectations.match_allowing_invocation(symbol, *arguments))
-        matching_expectation_allowing_invocation.invoke(&block)
+        matching_expectation_allowing_invocation.invoke(*arguments, &block)
```

Comment:
As an aside, I like the fact that the invocation is now encapsulated in an object and I wonder if it might now make sense for methods like `ExpectationList#match_allowing_invocation`, `ExpectationList#match` & `ExpectationList#matches_method?` to take a invocation object as their argument...? Also I wonder if there is now some duplication between `Invocation` and `UnexpectedInvocation`...? I've captured these thoughts in https://github.com/freerange/mocha/issues/408 and https://github.com/freerange/mocha/issues/409.

Reply:
Given that I've captured these thoughts in new issues, I'm going to mark this as resolved.